ledger/participant-state/kvutils/src/test/suite/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/validation/ModelConformanceValidatorSpec.scala

```diff
@@ -50,12 +95,164 @@ class ModelConformanceValidatorSpec extends AnyWordSpec with Matchers with Mocki
     }
 
   val Seq(tx1, tx2, txNone, txOther) = lookupNodes map { node =>
-    val builder = TransactionBuilder()
+    val builder = txBuilder
     val rootId = builder.add(exercise)
     val lookupId = builder.add(node, rootId)
     builder.build() -> lookupId
   }
 
+  private val transactionEntry1 = DamlTransactionEntrySummary(
+    DamlTransactionEntry.newBuilder
+      .setSubmissionSeed(aSubmissionSeed)
+      .setLedgerEffectiveTime(Conversions.buildTimestamp(ledgerEffectiveTime))
+      .setTransaction(Conversions.encodeTransaction(tx1._1))
+      .build
+  )
+
+  "createValidationStep" should {
+    "create StepContinue in case of correct input" in {
+      val mockValidationResult = mock[Result[Unit]]
+      when(
+        mockValidationResult.consume(
+          any[Value.ContractId => Option[
+            Value.ContractInst[Value.VersionedValue[Value.ContractId]]
+          ]],
+          any[Ref.PackageId => Option[Ast.Package]],
+          any[GlobalKeyWithMaintainers => Option[Value.ContractId]],
+        )
+      ).thenReturn(Right(()))
+      when(
+        mockEngine.validate(
+          any[Set[Ref.Party]],
+          any[SubmittedTransaction],
+          any[Timestamp],
+          any[Ref.ParticipantId],
+          any[Timestamp],
+          any[Hash],
+        )
+      ).thenReturn(mockValidationResult)
+
+      modelConformanceValidator.createValidationStep(rejections)(
+        createCommitContext(
+          None,
+          Map(
+            inputContractIdStateKey -> Some(makeContractIdStateValue()),
+            contractIdStateKey1 -> Some(makeContractIdStateValue()),
+          ),
+        ),
+        transactionEntry1,
+      ) shouldBe StepContinue(transactionEntry1)
+    }
+
+    "create StepStop in case of validation error" in {
+      when(
+        mockEngine.validate(
+          any[Set[Ref.Party]],
+          any[SubmittedTransaction],
+          any[Timestamp],
+          any[Ref.ParticipantId],
+          any[Timestamp],
+          any[Hash],
+        )
+      ).thenReturn(ResultError(LfError.Validation.ReplayMismatch(mkMismatch(tx1, tx2))))
+
+      val step = modelConformanceValidator
+        .createValidationStep(rejections)(
+          createCommitContext(
+            None,
+            Map(
+              inputContractIdStateKey -> Some(makeContractIdStateValue()),
+              contractIdStateKey1 -> Some(makeContractIdStateValue()),
+            ),
+          ),
+          transactionEntry1,
+        )
+      step shouldBe a[StepStop]
+      step
+        .asInstanceOf[StepStop]
+        .logEntry
+        .getTransactionRejectionEntry
+        .hasInconsistent shouldBe true
```

Comment:
As per my comment above, I think a `ReplayMismatch` error should resolve to a `Disputed` error.